projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/DeploymentModelConverter.java

```diff
@@ -37,26 +34,61 @@ public static JobDeployment toJobDeployment(
     return deployment;
   }
 
-  public static JobDeployment toJobDeployment(
-      String teamName, String jobName, DataJobDeployment jobDeploymentStatus) {
-    JobDeployment deployment = new JobDeployment();
-    deployment.setDataJobTeam(teamName);
-    deployment.setDataJobName(jobName);
+  public static DesiredDataJobDeployment toDesiredDataJobDeployment(
+      JobDeployment jobDeploymentStatus) {
+    DesiredDataJobDeployment deployment = new DesiredDataJobDeployment();
+    deployment.setDataJobName(jobDeploymentStatus.getDataJobName());
     deployment.setEnabled(jobDeploymentStatus.getEnabled());
 
-    DataJobResources dataJobResources = new DataJobResources();
-    dataJobResources.setCpuLimit(jobDeploymentStatus.getResourcesCpuLimit());
-    dataJobResources.setCpuRequest(jobDeploymentStatus.getResourcesCpuRequest());
-    dataJobResources.setMemoryLimit(jobDeploymentStatus.getResourcesMemoryLimit());
-    dataJobResources.setMemoryRequest(jobDeploymentStatus.getResourcesMemoryRequest());
-    deployment.setResources(dataJobResources);
+    DataJobResources dataJobResources = jobDeploymentStatus.getResources();
+
+    if (dataJobResources != null) {
+      DataJobDeploymentResources deploymentResources = new DataJobDeploymentResources();
+      deploymentResources.setCpuRequestCores(dataJobResources.getCpuRequest());
+      deploymentResources.setCpuLimitCores(dataJobResources.getCpuLimit());
+      deploymentResources.setMemoryRequestMi(dataJobResources.getMemoryRequest());
+      deploymentResources.setMemoryLimitMi(dataJobResources.getMemoryLimit());
+    }
 
     deployment.setGitCommitSha(jobDeploymentStatus.getGitCommitSha());
     deployment.setPythonVersion(jobDeploymentStatus.getPythonVersion());
 
     return deployment;
   }
 
+  public static ActualDataJobDeployment toActualJobDeployment(
+      DesiredDataJobDeployment desiredDataJobDeployment,
+      String deploymentVersionSha,
+      OffsetDateTime lastDeployedDate) {
+    ActualDataJobDeployment deployment = new ActualDataJobDeployment();
+    deployment.setDataJobName(desiredDataJobDeployment.getDataJobName());
+    deployment.setDataJob(desiredDataJobDeployment.getDataJob());
+    deployment.setEnabled(desiredDataJobDeployment.getEnabled());
+
+    DataJobDeploymentResources desiredDataJobDeploymentResources =
+        desiredDataJobDeployment.getResources();
+
+    if (desiredDataJobDeploymentResources != null) {
```

Comment:
extract duplicate code in a method - the same code is part of the previous method.